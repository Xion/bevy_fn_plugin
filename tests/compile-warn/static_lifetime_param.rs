// We actually want to fail on the lifetime warning, but that one is a force-warn
// category which cannot be affected by allow/deny directives.
// This is fine, however, because the warning will still be emitted regardless so trybuild
// can verify its presence.
#![deny(unused_imports)]

#[macro_use] extern crate bevy_fn_plugin;

use bevy::prelude::*;

#[bevy_plugin]
fn GenericPlugin<'a: 'static>(_: &mut App) {
    let _: Option<&'a ()> = None;
}

fn main() {}
