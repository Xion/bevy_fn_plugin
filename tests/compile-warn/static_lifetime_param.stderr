error: unused import: `bevy::prelude::*`
 --> tests/compile-warn/static_lifetime_param.rs:9:5
  |
9 | use bevy::prelude::*;
  |     ^^^^^^^^^^^^^^^^
  |
note: the lint level is defined here
 --> tests/compile-warn/static_lifetime_param.rs:5:9
  |
5 | #![deny(unused_imports)]
  |         ^^^^^^^^^^^^^^

warning: unnecessary lifetime parameter `'a`
  --> tests/compile-warn/static_lifetime_param.rs:12:18
   |
12 | fn GenericPlugin<'a: 'static>(_: &mut App) {
   |                  ^^
   |
   = help: you can use the `'static` lifetime directly, in place of `'a`
